File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/produto.dart
class Produto {
    int _codigo;
    String _nome;
    double _preco;
    int _quantidade;

    Produto(this._codigo, this._nome, this._preco, this._quantidade);

    int get codigo => _codigo;
    set codigo(int value) {
        _codigo = value;
    }

    String get nome => _nome;
    set nome(String value) {
        _nome = value;
    }

    double get preco => _preco;
    set preco(double value) {
        if (value >= 0) {
        _preco = value;
        }
    }

    int get quantidade => _quantidade;
    set quantidade(int value) {
        if (value >= 0) {
        _quantidade = value;
        }
    }

    double calcularPrecoTotal() {
        return _preco * _quantidade;
    }

    @override
    String toString() {
        return '($_codigo) $_nome: $_preco x $_quantidade = ${calcularPrecoTotal()}';
    }
}

void main() {
    Produto produto1 = Produto(1, "Teclado", 19.9, 2);
    print(produto1);

    Produto produto2 = Produto(2, "Mouse", 25.5, 3);
    print(produto2);

    produto1.quantidade = 5;
    produto1.preco = 20.0;

    print("Após alteração:");
    print(produto1);
}

File: 5 semestre/Programacao para Dispositivos Moveis/UAS/UA1/java/src/Produto.java
public class Produto {

    // Atributos privados
    private int codigo;
    private String nome;
    private double preco;
    private int quantidade;

    // Construtor
    public Produto(int codigo, String nome, double preco, int quantidade) {
        this.codigo = codigo;
        this.nome = nome;
        this.preco = preco;
        this.quantidade = quantidade;
    }

    // Getters e Setters
    public int getCodigo() {
        return codigo;
    }

    public void setCodigo(int codigo) {
        this.codigo = codigo;
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public double getPreco() {
        return preco;
    }

    public void setPreco(double preco) {
        if (preco >= 0) {
            this.preco = preco;
        }
    }

    public int getQuantidade() {
        return quantidade;
    }

    public void setQuantidade(int quantidade) {
        if (quantidade >= 0) {
            this.quantidade = quantidade;
        }
    }

    // Método para calcular preço total
    public double calcularPrecoTotal() {
        return preco * quantidade;
    }

    // Sobrescrita do toString
    @Override
    public String toString() {
        return "(" + codigo + ") " + nome + ": " + preco + " x " + quantidade + " = " + calcularPrecoTotal();
    }
}